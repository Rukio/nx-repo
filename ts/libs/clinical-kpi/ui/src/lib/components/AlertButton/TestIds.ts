export const ALERT_BUTTON_TEST_IDS = {
  TEXT: 'alert-button-text',
  BUTTON: 'alert-button-button',
};
