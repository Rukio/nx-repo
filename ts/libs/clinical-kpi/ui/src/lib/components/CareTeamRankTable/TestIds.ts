export const CARE_TEAM_RANK_TABLE_TEST_IDS = {
  TABLE: 'care-team-rank-table',
  getRankTestId: (rank?: number) => `care-team-table-rank-${rank}`,
  getProviderNameTestId: (rank?: number) =>
    `care-team-rank-table-provider-name-${rank}`,
  getValueTestId: (rank?: number) => `care-team-rank-table-value-${rank}`,
  getValueChangeTestId: (rank?: number) =>
    `care-team-rank-table-value-change-${rank}`,
  getPlaceholderTestId: (rank?: number) =>
    `care-team-rank-table-placeholder-${rank}`,
};

export const CARE_TEAM_PLACEHOLDER_ROW_RANKS = [1, 2, 3, 4, 5, 6, 7];
