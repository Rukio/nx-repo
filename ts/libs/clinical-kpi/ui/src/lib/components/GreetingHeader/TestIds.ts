export const GREETING_HEADER_TEST_IDS = {
  GREETING_TEXT: 'header-greeting-text',
  LAST_UPDATED_TEXT: 'header-last-updated-text',
  VISITS_COMPLETED_TEXT: 'header-visits-completed-text',
  PERFORMANCE_TEXT: 'header-performance-text',
  LEARN_MORE_LINK: 'header-learn-more-link',
  LAST_UPDATED_PLACEHOLDER: 'header-last-updated-placeholder',
  VISITS_COMPLETED_PLACEHOLDER: 'header-visits-completed-placeholder',
  STATION_LINK: 'header-station-dashboard-link',
  STATION_LINK_LABEL_PREFIX: 'header-station-dashboard-link-label-prefix',
};
