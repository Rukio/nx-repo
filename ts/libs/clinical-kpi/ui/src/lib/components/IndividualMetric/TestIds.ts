export const INDIVIDUAL_METRIC_TEST_IDS = {
  TYPE: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-type`,
  VALUE: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-value`,
  VALUE_CHANGE: (testIdPrefix: string) =>
    `${testIdPrefix}-individual-metric-change`,
  PERFORMANCE: (testIdPrefix: string) =>
    `${testIdPrefix}-individual-metric-performance`,
  GOAL: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-goal`,
};
