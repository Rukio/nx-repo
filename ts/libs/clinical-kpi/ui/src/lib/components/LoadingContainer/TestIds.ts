export const LOADING_CONTAINER_TEST_IDS = {
  SPINNER: (prefix: string) => `${prefix}-loading`,
};
