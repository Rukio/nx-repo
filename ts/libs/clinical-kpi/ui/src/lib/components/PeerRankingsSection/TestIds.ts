export const PEER_RANKINGS_SECTION_TEST_IDS = {
  TITLE: 'peer-rankings-section-title',
};
