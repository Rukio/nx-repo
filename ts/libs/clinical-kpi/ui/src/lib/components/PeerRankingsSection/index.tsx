export { default as PeerRankingsSection } from './PeerRankingsSection';
export { PEER_RANKINGS_SECTION_TEST_IDS } from './TestIds';
