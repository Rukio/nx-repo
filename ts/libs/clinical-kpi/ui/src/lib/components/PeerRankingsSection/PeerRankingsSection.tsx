import { FC, PropsWithChildren } from 'react';
import { Typography, makeSxStyles, Box } from '@*company-data-covered*/design-system';
import { PEER_RANKINGS_SECTION_TEST_IDS } from './TestIds';

export type PeerRankingsSectionProps = PropsWithChildren<{
  title: string;
}>;

const makeStyles = () =>
  makeSxStyles({
    sectionHeader: {
      fontWeight: 600,
    },
    childrenSection: {
      flexGrow: 1,
      mt: 2,
    },
  });

const PeerRankingsSection: FC<PeerRankingsSectionProps> = ({
  title,
  children,
}) => {
  const styles = makeStyles();

  return (
    <>
      <Typography
        variant="body2"
        data-testid={PEER_RANKINGS_SECTION_TEST_IDS.TITLE}
        sx={styles.sectionHeader}
      >
        {title}
      </Typography>
      <Box sx={styles.childrenSection}>{children}</Box>
    </>
  );
};

export default PeerRankingsSection;
