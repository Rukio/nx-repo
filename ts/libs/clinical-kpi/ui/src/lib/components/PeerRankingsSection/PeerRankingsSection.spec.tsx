import { render, screen } from '@testing-library/react';

import PeerRankingsSection, {
  PeerRankingsSectionProps,
} from './PeerRankingsSection';
import { PEER_RANKINGS_SECTION_TEST_IDS } from './TestIds';

const peerRankingsSectionProps: PeerRankingsSectionProps = {
  title: 'Title',
};

const fakeChildProps = {
  dataTestId: 'fakeChild',
  textContent: 'fakeChild',
};

describe('PeerRankingsSection', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <PeerRankingsSection {...peerRankingsSectionProps}>
        <div data-testid={fakeChildProps.dataTestId}>
          {fakeChildProps.textContent}
        </div>
      </PeerRankingsSection>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(
      screen.getByTestId(PEER_RANKINGS_SECTION_TEST_IDS.TITLE)
    ).toHaveTextContent(peerRankingsSectionProps.title);
    expect(screen.getByTestId(fakeChildProps.dataTestId)).toHaveTextContent(
      fakeChildProps.textContent
    );
  });
});
