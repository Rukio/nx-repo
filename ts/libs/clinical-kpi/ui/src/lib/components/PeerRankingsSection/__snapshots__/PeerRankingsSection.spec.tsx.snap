// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeerRankingsSection should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  font-weight: 600;
}

<p
    class="MuiTypography-root MuiTypography-body2 emotion-0"
    data-testid="peer-rankings-section-title"
  >
    Title
  </p>
  .emotion-0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-top: 16px;
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      data-testid="fakeChild"
    >
      fakeChild
    </div>
  </div>
</DocumentFragment>
`;
